### DISCUSSION TITLE
Cannot disable a default keymap for leap.nvim

### DISCUSSION THREAD

**Discussion by krishna-bala** [LazyVim/LazyVim#584] [Q&A] [answered]
In [LazyVim - Adding & Disabling Plugin Keymaps](https://www.lazyvim.org/configuration/plugins#%EF%B8%8F-adding--disabling-plugin-keymaps) the following instruction is given:

> You can also disable a default keymap by setting it to false. To override a keymap, simply add one with the same lhs and a new rhs.
```
lua/plugins/telescope.lua
```
```
{
  "nvim-telescope/telescope.nvim",
  keys = {
    -- disable the keymap to grep files
    {"<leader>/", false},
    -- change a keymap
    { "<leader>ff", "<cmd>Telescope find_files<cr>", desc = "Find Files" },
    -- add a keymap to browse plugin files
    {
      "<leader>fp",
      function() require("telescope.builtin").find_files({ cwd = require("lazy.core.config").options.root }) end,
      desc = "Find Plugin File",
    },
  },
},
```

However, I am having issues disabling a default keymap for the `leap.nvim` plugin. Here is what I have:
```
lua/plugins/leap.lua
```
```
return {
  "ggandor/leap.nvim",
  keys = {
    { "gs", false },
  },
}
```

After restarting my nvim instance, I still see "gs" in my keymaps and it still functions according to the default keymapping. Am I missing something obvious here? My apologies if this has already been asked in some form, but I searched the issues and discussions and wasn't able to find any suggestion that works. I have also tried replacing "false" with "vim.NIL", but that did not work either.

**Comment by dpetka2001** [comment, marked as answer]
There was a similar problem and the maintainer proposed this [solution](https://github.com/LazyVim/LazyVim/discussions/76#discussioncomment-5157196). Check if it works for you.

**Reply by dpetka2001** [reply]
You could also check [this](https://github.com/LazyVim/LazyVim/discussions/90) topic which has similar solutions.

**Reply by krishna-bala** [reply]
Thank you for the quick response! This worked for me. For reference, here's what I did:
```
-- lua/plugins/leap.lua
return {
  "ggandor/leap.nvim",
  keys = false,
  config = function(_, opts)
    -- add your own keymaps here
    vim.keymap.set({ "n", "x", "o" }, "s", "<Plug>(leap-forward-to)", { desc = "leap-forward-to" })
    vim.keymap.set({ "n", "x", "o" }, "S", "<Plug>(leap-backward-to)", { desc = "leap-backward-to" })
    vim.keymap.set({ "n", "x", "o" }, ";gs", "<Plug>(leap-from-window)", { desc = "leap-from-window" })
  end,
}
```
